### PR TITLE
update to latest pyo3 and numpy versions, adjusted function signatures

### PR DESCRIPTION
**Pull request by g-bauer**

### CODE REVIEW

File: build_wheel/src/lib.rs [hunk 1]
```diff
@@ -1,7 +1,7 @@
-use num_dual as num_dual_rs;
+use ::num_dual as num_dual_rs;
 use pyo3::prelude::*;
 
-/// Implementation of SI numbers.
+/// Implementation of dual numbers.
```
Comment (prehner) [inline build_wheel/src/lib.rs:4]: 🙈

File: src/python/macros.rs [hunk 2]
```diff
@@ -213,7 +213,7 @@ macro_rules! impl_dual_num {
             }
 
             #[inline]
-            #[pyo3(text_signature = "($self, b: Self, c: Self)")]
+            #[pyo3(text_signature = "($self, a, b)")]
```
Comment (prehner) [inline src/python/macros.rs:216]: wouldn't this be auto generated?

File: src/python/hyperdual.rs [hunk 3]
```diff
@@ -113,7 +113,7 @@ macro_rules! impl_hyper_dual_mn {
 macro_rules! impl_derive {
     ([$(($py_type_name:ident, $n:literal)),+; $(($py_type_name12:ident, $py_type_name21:ident, $m:literal)),+; $(($py_type_name3:ident, $m1:literal, $m2:literal)),+]) => {
         #[pyfunction]
-        #[pyo3(text_signature = "(x1, x2=None)")]
+        #[pyo3(signature = (x1, x2=None), text_signature = "(x1, x2=None)")]
```
Comment (prehner) [inline src/python/hyperdual.rs:116]: is the `signature` necessary?

File: src/python/macros.rs (hunk 2 above)
Comment (g-bauer) [reply src/python/macros.rs]: Wasn't sure about the `$self` but yes - it is auto generated.

File: src/python/hyperdual.rs (hunk 3 above)
Comment (g-bauer) [reply src/python/hyperdual.rs]: Turns out, it's not.

**Review by prehner — APPROVED** [review]

**Pull request merged**